--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -389,9 +389,9 @@
   else()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m64")
     set(MKL_LIBRARIES
-      ${MKL_LIBRARY_DIR}/libmkl_core.a
-      ${MKL_LIBRARY_DIR}/libmkl_intel_ilp64.a
+      ${MKL_LIBRARY_DIR}/libmkl_core.so
+      ${MKL_LIBRARY_DIR}/libmkl_intel_ilp64.so
       )
   endif()

   if(OPENMP_RUNTIME STREQUAL "INTEL")
@@ -401,14 +401,14 @@
       list(APPEND MKL_LIBRARIES ${MKL_LIBRARY_DIR}/libmkl_intel_thread.a)
     endif()
   elseif(OPENMP_RUNTIME STREQUAL "COMP")
     if(WIN32)
       message(FATAL_ERROR "Building with MKL requires Intel OpenMP")
     else()
-      list(APPEND MKL_LIBRARIES ${MKL_LIBRARY_DIR}/libmkl_gnu_thread.a)
+      list(APPEND MKL_LIBRARIES ${MKL_LIBRARY_DIR}/libmkl_gnu_thread.so)
     endif()
   elseif(OPENMP_RUNTIME STREQUAL "NONE")
     if(WIN32)
       list(APPEND MKL_LIBRARIES ${MKL_LIBRARY_DIR}/mkl_sequential.lib)
     else()
-      list(APPEND MKL_LIBRARIES ${MKL_LIBRARY_DIR}/libmkl_sequential.a)
+      list(APPEND MKL_LIBRARIES ${MKL_LIBRARY_DIR}/libmkl_sequential.so)
     endif()
   endif()
